adoc/chapters/programming_interface.adoc

```diff
@@ -963,207 +963,237 @@ are handled by throwing synchronous SYCL exceptions.
+_Constraints:_ The [code]#DeviceSelector# must be a type that satisfies the
+requirements of a <<device-selector>> as defined in <<sec:device-selector>>.
 
+_Effects:_ The [code]#selector# is called for every <<root-device>> as described
+in <<sec:device-selector>>.
+Constructs a [code]#platform# object that is a copy of the platform which
+contains the device that is selected by [code]#selector#.
 
+'''
 
-[[table.members.platform]]
-.Member functions of the SYCL [code]#platform# class
-[width="100%",options="header",separator="@",cols="65%,35%"]
-|====
-@ Member function @ Description
-a@
-[source]
+[[sec:platform-member-funcs]]
+==== Member functions
+
+.[apidef]#platform::get_backend#
+[source,role=synopsis,id=api:platform-get-backend]
 ----
 backend get_backend() const noexcept
 ----
-   a@ Returns a [code]#backend# identifying the <<backend>> associated
-      with this [code]#platform#.
 
-a@
-[source]
+_Returns:_ The <<backend>> that is associated with this platform.
+
+'''
+
+.[apidef]#platform::get_info#
+[source,role=synopsis,id=api:platform-get-info]
 ----
-template <typename Param> typename Param::return_type get_info() const
+template <typename Param>
+typename Param::return_type get_info() const
 ----
-   a@ Queries this SYCL [code]#platform# for information requested by
-      the template parameter [code]#Param#.
-      The type alias [code]#Param::return_type# must be defined in
-      accordance with the info parameters in <<table.platform.info>> to
-      facilitate returning the type associated with the [code]#Param#
-      parameter.
 
-a@
-[source]
+_Constraints:_ The [code]#Param# must be an information descriptor for the
+platform class.
+
+Each information descriptor specifies the return value and may also specify
+preconditions, exceptions that are thrown, etc.
+
+'''
+
+.[apidef]#platform::get_backend_info#
+[source,role=synopsis,id=api:platform-get-backend-info]
 ----
-template <typename Param> typename Param::return_type get_backend_info() const
+template <typename Param>
+typename Param::return_type get_backend_info() const
 ----
-   a@ Queries this SYCL [code]#platform# for <<backend>>-specific
-      information requested by the template parameter [code]#Param#.
-      The type alias [code]#Param::return_type# must be defined in
-      accordance with the <<backend>> specification.
-      Must throw an [code]#exception# with the [code]#errc::backend_mismatch#
-      error code if the <<backend>> that corresponds with [code]#Param# is different
-      from the <<backend>> that is associated with this [code]#platform#.
 
-a@
-[source]
+_Constraints:_ The [code]#Param# must be a backend information descriptor for
+the platform class.
+
+_Throws:_ An [code]#exception# with the [code]#errc::backend_mismatch# error
+code if the backend that corresponds with [code]#Param# is different from the
+backend that is associated with this platform.
+
+Each information descriptor specifies the return value and may also specify
```

Comment:
Do we want to add a hyperlink to the information descriptors?

Reply:
Yes, that's a good idea.  Done in 6915b81636ef42e0333f24aff19c08dfd62e136e.